pyaedt/modules/SetupTemplates.py

```diff
@@ -2172,3 +2173,40 @@ def __init__(self, setup, props):
 
     def _setitem_without_update(self, key, value):
         OrderedDict.__setitem__(self, key, value)
+
+    def _export_properties_to_json(self, fpath):
+        """Export all setup properties into a json file.
+
+        Parameters
+        ----------
+        fpath : str
+            File path of the json file.
+        """
+        if not fpath.endswith(".json"):
+            fpath = fpath + ".json"
+        with open(fpath, "w", encoding="utf-8") as f:
```

Comment:
@ring630 Do we want to override the file if it already exists? Is it ok? Or shall we raise an exception?

Reply:
@MaxJPRey the user may or may not want to overwrite the existing file. We can add an argument overwrite=False